### PR TITLE
MVP Processors Preview: Proxy code split with named bundle.

### PR DESCRIPTION
**Pull request by Chris-V**

### CODE REVIEW

File: gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/client/CodeSplitNamedBundleProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2011 ArcBees Inc.
```
Comment (olafleur) [inline gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/client/CodeSplitNamedBundleProvider.java]: 2011 ?

**Comment by olafleur** [comment]
LGTM

File: gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/NamedProviderBundleCollectionProcessor.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        for (Entry<String, List<Type>> bundle : metaInfReader.getBundles().entrySet()) {
+            for (Type type : bundle.getValue()) {
+                collection.addBundle(bundle.getKey(), type);
+            }
+        }
+    }
+
+    private void addBundlesToCollection(Set<? extends Element> bundles) {
+        for (Element bundle : bundles) {
+
```
Comment (meriouma) [inline gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/NamedProviderBundleCollectionProcessor.java]: nl

File: gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/ProviderBundleMetaInfReader.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            loadBundlesFromZip(zipFile);
+        }
+    }
+
+    private void loadBundlesFromZip(ZipFile file) throws IOException {
+        Enumeration<? extends ZipEntry> entries = file.entries();
+
+        while (entries.hasMoreElements()) {
+            ZipEntry entry = entries.nextElement();
+            if (!entry.isDirectory() && entry.getName().startsWith("META-INF/gwtp/bundles")) {
```
Comment (meriouma) [inline gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/ProviderBundleMetaInfReader.java]: Could be extracted to a constant

File: gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/ProviderBundleMetaInfWriter.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+    private BufferedWriter getOrCreateWriter(String bundleName) throws IOException {
+        BufferedWriter writer;
+
+        if (writers.containsKey(bundleName)) {
+            writer = writers.get(bundleName);
+        } else {
+            FileObject file = outputter.prepareSourceFile(new Type("", "gwtp/bundles/" + bundleName), META_INF);
```
Comment (meriouma) [inline gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/processors/bundles/ProviderBundleMetaInfWriter.java]: gwtp/bundles again :P

File: gwtp-core/gwtp-clients-common/src/main/resources/com/gwtplatform/common/processors/bundles/NamedProviderBundleAggregate.vm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import com.gwtplatform.common.client.NamedProviderBundle;
+#printImports($imports)
+
+@Generated(value = "$processor.qualifiedName", date = "$processingDate")
+public class $impl.simpleName extends NamedProviderBundle {
+    private final List<NamedProviderBundle> bundles = new ArrayList<>();
+
+    @Inject
+    ${impl.simpleName}(#foreach($bundle in $bundles)
+
```
Comment (meriouma) [inline gwtp-core/gwtp-clients-common/src/main/resources/com/gwtplatform/common/processors/bundles/NamedProviderBundleAggregate.vm]: Extra nl?

File: gwtp-core/gwtp-clients-common/src/main/resources/com/gwtplatform/common/processors/bundles/NamedProviderBundleAggregate.vm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#foreach($bundle in $bundles)
+        bundles.add(bundle$foreach.index);
+#end
+    }
+
+    @Override
+    public Provider<?> get(String name) {
+        for (NamedProviderBundle bundle : bundles) {
+            Provider<?> provider = bundle.get(name);
+            if(provider != null) {
```
Comment (meriouma) [inline gwtp-core/gwtp-clients-common/src/main/resources/com/gwtplatform/common/processors/bundles/NamedProviderBundleAggregate.vm]: Missing space after if :P

**Comment by meriouma** [comment]
LGTM. It seems the $$ is there again at a few places

**Pull request merged**